orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/cloudformation/ExecuteCloudFormationChangeSetTask.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Adevinta.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.cloudformation;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class ExecuteCloudFormationChangeSetTask extends AbstractCloudProviderAwareTask
+    implements Task {
+
+  @Autowired KatoService katoService;
+
+  public static final String TASK_NAME = "executeCloudFormationChangeSet";
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+
+    String actionOnReplacement =
+        (String) Optional.ofNullable(stage.getContext().get("actionOnReplacement")).orElse("fail");
+
+    Optional<Map> currentChangeSet = getCurrentChangeSet(stage);
+    if (currentChangeSet.isPresent()) {
+      if (isAnyChangeSetReplacement(currentChangeSet.get())) {
+        switch (actionOnReplacement) {
+          case "skip":
```

Comment:
These actions should probably be constants somewhere, so the constant can be reused in other tasks and we don't risk breaking the workflow if we decide to change the value of the replacement string.

Reply:
Thank you, good point. I've moved those values to constants.